examples/hybrid/callbacks.jl

```diff
@@ -83,23 +63,58 @@ function get_callbacks(parsed_args, simulation, model_spec, params)
     )
 end
 
+function call_every_n_steps(f!, n = 1; skip_first = false, call_at_end = false)
+    previous_step = Ref(0)
+    return ODE.DiscreteCallback(
+        (u, t, integrator) ->
+            (previous_step[] += 1) % n == 0 ||
+                (call_at_end && t == integrator.sol.prob.tspan[2]),
+        f!;
+        initialize = (cb, u, t, integrator) -> skip_first || f!(integrator),
+        save_positions = (false, false),
+    )
+end
 
-condition_every_iter(u, t, integrator) = true
+function call_every_dt(f!, dt; skip_first = false, call_at_end = false)
+    next_t = Ref{typeof(dt)}()
+    affect! = function (integrator)
+        t = integrator.t
+        t_end = integrator.sol.prob.tspan[2]
+        while t >= next_t[]
+            f!(integrator)
+            next_t[] =
+                (call_at_end && t < t_end) ? min(t_end, next_t[] + dt) :
+                next_t[] + dt
+        end
+    end
+    return ODE.DiscreteCallback(
+        (u, t, integrator) -> t >= next_t[],
+        affect!;
+        initialize = (cb, u, t, integrator) -> begin
+            skip_first || f!(integrator)
+            t_end = integrator.sol.prob.tspan[2]
+            next_t[] =
+                (call_at_end && t < t_end) ? min(t_end, t + dt) : t + dt
+        end,
+        save_positions = (false, false),
+    )
+end
```

Comment:
What is this doing? Why do we need the Ref here?

Reply:
Well we need some way of accessing the next stop time from the various closures. Is it better to have next_t be a Ref (so that it doesn't change but the value in it does change), or to have it be a regular Float (so that it actually changes)? I was under the impression that closures which use variables that change might run into performance issues.